[OSDOCS-6942]:OSD on GCP can be installed into Shared VPC (XPN) (#66312)
openshift/openshift-docs: modules/osd-create-cluster-ccs.adoc

+
.. If you are installing into an existing VPC and you want to enable an HTTP or HTTPS proxy for your cluster, select *Configure a cluster-wide proxy*.
endif::osd-on-aws[]

ifdef::osd-on-gcp[]
. Optional: To install the cluster in an existing GCP Virtual Private Cloud (VPC):
.. Select *Install into an existing VPC*.
.. If you are installing into an existing VPC and you want to enable an HTTP or HTTPS proxy for your cluster, select *Configure a cluster-wide proxy*.
endif::osd-on-gcp[]
+
. Click *Next*.

ifdef::osd-on-gcp[]
. Optional: To install the cluster into a GCP shared VPC:
+
[IMPORTANT]
====

> 

Reviewer: Remove the extra line.

> To install a cluster into a shared VPC, you must use {product-title} version 4.13.15 or above. Additionally, the shared VPC administrator must enable a project as a host project in their Google Cloud console. For more information, see link:https://cloud.google.com/vpc/docs/provisioning-shared-vpc#set-up-shared-vpc[Enable a host project].

Reviewer: ```suggestion
To install a cluster into a shared VPC, you must use {product-title} version 4.13.15 or above. Additionally, the Shared VPC Admin must enable a project as a host project in their Google Cloud console. For more information, see link:https://cloud.google.com/vpc/docs/provisioning-shared-vpc#set-up-shared-vpc[Enable a host project].
```

====

.. Select *Install into GCP shared VPC*.

> .. Specify the **Host project ID**. If the specified host project ID is incorrect, cluster creation will fail.

Reviewer: Is "host project ID" initial capped or not? You have it written two different ways here.

Reviewer 2: More on this, since it's bold, I am assuming that it's a field name. Is that correct?  If so, I think the rest is fine, though we may want to mirror how it displays throughout the UI.

We tend to avoid future tense too, so maybe "cluster creation fails."

Author: Yes, it is field name. Thanks all! 
![image](https://github.com/openshift/openshift-docs/assets/122639474/471488aa-8b5a-442e-a1bd-22a3b6d97ac3)

+

> 

Reviewer: I don't think the extra line is necessary here.

[IMPORTANT]
====

> Once you complete the steps within the cluster configuration wizard and click **Create Cluster**, the cluster will go into the "Installation Waiting" state. At this point, you must contact the VPC administrator of the host project, who must assign the dynamically-generated service account the following roles: **Computer Network Administrator**, **Compute Security Administrator**, and **DNS Administrator**.

Reviewer: ```suggestion
Once you complete the steps within the cluster configuration wizard and click *Create Cluster*, the cluster will go into the "Installation Waiting" state. At this point, you must contact the Shared VPC Admin, who must assign the dynamically-generated service account the following roles: *Computer Network Administrator*, *Compute Security Administrator*, and *DNS Administrator*.
```

> The administrator has 30 days to grant the listed permissions before the cluster creation fails.

Reviewer: ```suggestion
The Shared VPC Admin has 30 days to grant the listed permissions before cluster creation fails.
```

> For information about shared GPC VPC permissions, see link:https://cloud.google.com/vpc/docs/provisioning-shared-vpc#migs-service-accounts[Provision shared VPC].

Reviewer: ```suggestion
For information about GCP shared VPC permissions, see link:https://cloud.google.com/vpc/docs/provisioning-shared-vpc#migs-service-accounts[Provision Shared VPC].
```
The Google Docs title cap their headings, so I wanted it to be consistent.

====
endif::osd-on-gcp[]
+
. If you opted to install the cluster in an existing
ifdef::osd-on-aws[]
AWS
endif::osd-on-aws[]
ifdef::osd-on-gcp[]
GCP
endif::osd-on-gcp[]

> VPC, provide your *Virtual Private Cloud (VPC) subnet settings* and select *Next*.

Reviewer: ```suggestion
VPC, provide your *VPC subnet settings* and select *Next*.
```
Not sure why VPC was spelled out all the way down here. There are other, better "first mentions".

Author: ![image](https://github.com/openshift/openshift-docs/assets/122639474/01f67332-984b-4476-bc3d-197a2df283d1)
Good point. FWIW, I think it was originally written this way b/c of the way it was presented in the UI.

====

. Review the summary of your selections and click *Create cluster* to start the cluster installation. The installation takes approximately 30-40 minutes to complete.
+
ifdef::osd-on-gcp[]
[NOTE]
====

> If you delete a cluster that was installed into a GCP shared VPC, you must inform the VPC owner of the host project to remove the service account and the roles granted to the service account that was referenced during the creation of the cluster.

Reviewer 3: In fact, the service account itself would be deleted during the cluster deletion. So suggest to put it like below, 

... you must inform the VPC owner of the host project to remove the IAM policy binding of the service account and its binding roles in the host project.

Author: Hello @svmrh,
For point #1, I have added a note to that doc: 
![image](https://github.com/openshift/openshift-docs/assets/122639474/e7c57191-4fb6-45db-965d-a73739f86747)

Author: For point #2, I have added a note: 
![image](https://github.com/openshift/openshift-docs/assets/122639474/97ce29c4-cf71-4bd2-b66c-edb02320bb55)

Author: > In fact, the service account itself would be deleted during the cluster deletion. So suggest to put it like below,
> 
> ... you must inform the VPC owner of the host project to remove the IAM policy binding of the service account and its binding roles in the host project.

Thanks @jianli-wei,
   I applied your suggestion with just a slight tweak in wording to fit the narrative prior to that note. Thanks for the clarity.   
![image](https://github.com/openshift/openshift-docs/assets/122639474/bb4a7c3f-64f5-46f3-a5b5-90eb2d38c536)

====

. Review the summary of your selections and click *Create cluster* to start the cluster installation. The installation takes approximately 30-40 minutes to complete.
+
ifdef::osd-on-gcp[]
[NOTE]
====

> If you delete a cluster that was installed into a GCP shared VPC, you must inform the VPC owner of the host project to remove the IAM policy roles granted to the service account that was referenced during the creation of the cluster.

Reviewer: Make this note match the corrections to the same note above.
